src/clj_jargon/permissions.clj

```diff
@@ -267,12 +294,9 @@
      (set-permissions cm user fpath read? write? own? false))
   ([cm user fpath read? write? own? recursive?]
      (validate-path-lengths fpath)
-     (cond
-      (item/is-file? cm fpath)
-      (set-dataobj-perms cm user fpath read? write? own?)
-
-      (item/is-dir? cm fpath)
-      (set-coll-perms cm user fpath read? write? own? recursive?))))
+     (case (item/object-type cm fpath)
```

Comment:
This is a case where using `case` is an improvement (though only for directories), since it results in only one call to `object-type` (and the underlying `getObjStat` call) rather than one to check it's a file, get false, then a second to check if it's a directory.